lib/howl/interactions/file_selection.moon

```diff
@@ -67,7 +60,18 @@ class FileSelector
 
   _chdir: (directory, text) =>
     @directory = directory
-    matcher = file_matcher self.directory_reader(directory), directory, @opts.allow_new
+    local matcher
+    if @show_subtree
+      items, timed_out = self.subtree_reader(directory, timeout: 1)
```

Comment:
1 second feels like a tight timeout to me, when switching to a recursive listing I would be willing to wait longer than getting truncated listings. This would be subjective of course, but how about 2 or 3 seconds?